exporter/azuremonitorexporter/traceexporter.go

```diff
@@ -0,0 +1,660 @@
+		data.Target = httpHost
+	} else if httpScheme != "" && peerHostName != "" && peerPort != "" && httpTarget != "" {
+		sb.WriteString(httpMethod)
+		sb.WriteString(" ")
+		sb.WriteString(httpTarget)
+		data.Name = sb.String()
+
+		sb.Reset()
+		sb.WriteString(httpScheme)
+		sb.WriteString("://")
+		sb.WriteString(peerHostName)
+		sb.WriteString(":")
+		sb.WriteString(peerPort)
+		sb.WriteString(httpTarget)
+		data.Data = sb.String()
+
+		sb.Reset()
+		sb.WriteString(peerHostName)
+		sb.WriteString(":")
+		sb.WriteString(peerPort)
+		data.Target = sb.String()
+	} else if httpScheme != "" && peerIP != "" && peerPort != "" && httpTarget != "" {
+		sb.WriteString(httpMethod)
+		sb.WriteString(" ")
+		sb.WriteString(httpTarget)
+		data.Name = sb.String()
+
+		sb.Reset()
+		sb.WriteString(httpScheme)
+		sb.WriteString("://")
+		sb.WriteString(peerIP)
+		sb.WriteString(":")
+		sb.WriteString(peerPort)
+		sb.WriteString(httpTarget)
+		data.Data = sb.String()
+
+		sb.Reset()
+		sb.WriteString(peerIP)
+		sb.WriteString(":")
+		sb.WriteString(peerPort)
+		data.Target = sb.String()
+	}
+}
+
+// Sets properties on an AppInsights RemoteDependencyData from the wire format Span when the outbound Span type is gRPC
+func fillRemoteDependencyDataGrpc(span *tracepb.Span, data *contracts.RemoteDependencyData) {
+	data.Type = "Grpc"
+
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
+	peerService := ""
+	peerHostName := ""
+	peerPort := ""
+
+	for k, v := range span.Attributes.AttributeMap {
+		stringValue := attributeValueAsString(v)
+		switch k {
+		case spanAttributeKeyComponent:
+			// do nothing
+		case spanAttributeKeyRPCStatusCode:
+			data.ResultCode = stringValue
+			code, err := strconv.Atoi(data.ResultCode)
+			if err == nil {
+				data.Success = code == int(codes.OK)
+			}
+		case spanAttributeKeyPeerService:
+			peerService = stringValue
+		case spanAttributeKeyPeerHostname:
+			peerHostName = stringValue
+		case spanAttributeKeyPeerPort:
+			peerPort = stringValue
+		default:
+			// Anything not on this list above ends up as a custom property
+			data.Properties[k] = stringValue
+		}
+	}
+
+	var sb strings.Builder
+	if peerService != "" && peerHostName != "" && peerPort != "" {
+		sb.WriteString(peerHostName)
+		sb.WriteString(":")
+		sb.WriteString(peerPort)
+		data.Target = sb.String()
+
+		// append the service name to the host:port
+		sb.WriteString(peerService)
+		data.Data = sb.String()
+	}
+}
+
+func prefixIfNecessary(s string, prefix string) string {
+	if strings.HasPrefix(s, prefix) {
+		return s
+	}
+
+	return prefix + s
+}
+
+func sanitize(sanitizeFunc func() []string) {
+	sanitizeWithCallback(sanitizeFunc, nil)
+}
+
+func sanitizeWithCallback(sanitizeFunc func() []string, warningCallback func(string)) {
+	sanitizeWarnings := sanitizeFunc()
+	for _, warning := range sanitizeWarnings {
+		if warningCallback == nil {
+			// TODO error handling
+			fmt.Println(warning)
```

Comment:
It would better to pass the logger here and use it instead of printing.

Reply:
I'll see what I can do here. The zap logger has to plumbed through. Feels like there should have been an interface facade for that thing...